polygeist/tools/cgeist/Test/Verification/sycl/constructors.cpp

```diff
@@ -87,10 +87,12 @@
 // CHECK-LLVM-SAME:                          ptr noundef byval([[RANGE_TYPE:%"class.sycl::_V1::range.1"]]) align 8 [[ARG1:%.*]]) #[[FUNCATTRS:[0-9]+]]
 // CHECK-LLVM-DAG: [[RANGE:%.*]] = alloca [[RANGE_TYPE]]
 // CHECK-LLVM-DAG: [[ID:%.*]] = alloca [[ID_TYPE]]
-// CHECK-LLVM: [[ID1_AS:%.*]] = addrspacecast ptr [[ID]] to ptr addrspace(4)
-// CHECK-LLVM: call spir_func void @_ZN4sycl3_V12idILi1EEC1ERKS2_(ptr addrspace(4) [[ID1_AS]], 
-// CHECK-LLVM: [[RANGE1_AS:%.*]] = addrspacecast ptr [[RANGE]] to ptr addrspace(4)
-// CHECK-LLVM: call spir_func void @_ZN4sycl3_V15rangeILi1EEC1ERKS2_(ptr addrspace(4) [[RANGE1_AS]],
+// CHECK-LLVM: [[ID_AS:%.*]] = addrspacecast ptr [[ID]] to ptr addrspace(4)
+// CHECK-LLVM: [[ARG0_AS:%.*]] = addrspacecast ptr [[ARG0]] to ptr addrspace(4)
+// CHECK-LLVM: [[TMP:%.*]] = alloca [[ID_TYPE]]
+// CHECK-LLVM: [[TMP_AS:%.*]] = addrspacecast ptr [[TMP]] to ptr addrspace(4)
+// CHECK-LLVM: call void @llvm.memcpy.p4.p4.i64(ptr addrspace(4) [[TMP_AS]], ptr addrspace(4) [[ARG0_AS]], i64 8, i1 false)
+// CHECK-LLVM: call void @llvm.memcpy.p4.p4.i64(ptr addrspace(4) [[ID_AS]], ptr addrspace(4) [[TMP_AS]], i64 8, i1 false) 
```

Comment:
I'm confused by this test. Before, it was weird that it didn't check the uses of `ARG0`/`ARG1`. You are checking `ARG0` is copied from, but you seem to test anything about the range.

Reply:
Updated test. You're right. Also, MLIR test now checks what we want to check